engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java

```diff
@@ -61,32 +95,312 @@ public void setActionsCacheManager(ActionsCacheManager actionsCacheManager) {
+
+        StringBuilder query = new StringBuilder("from org.hawkular.alerts.engine.impl.ispn.model.IspnAction where");
+        query.append(String.format(" tenantId = '%s'", tenantId));
+
+        if (filter) {
+            if (criteria.hasCTimeCriteria()) {
+                if (criteria.hasRangeCriteria()) {
+                    query.append(String.format(" and ctime between %d and %d", criteria.getStartTime(),
+                            criteria.getEndTime()));
+                } else if (criteria.hasStartCriteria()) {
+                    query.append(String.format(" and ctime >= %d", criteria.getStartTime()));
+                } else {
+                    query.append(String.format(" and ctime <= %d", criteria.getEndTime()));
+                }
+            }
+            if (criteria.hasActionIdCriteria()) {
+                Set<String> actionIds = new HashSet<>();
+                if (null != criteria.getActionIds()) {
+                    actionIds.addAll(criteria.getActionIds());
+                }
+                if (null != criteria.getActionId()) {
+                    actionIds.add(criteria.getActionId());
+                }
+
+                query.append(inClause("actionId", actionIds));
+            }
+            if (criteria.hasActionPluginCriteria()) {
+                Set<String> actionPlugins = new HashSet<>();
+                if (null != criteria.getActionPlugins()) {
+                    actionPlugins.addAll(criteria.getActionPlugins());
+                }
+                if (null != criteria.getActionPlugin()) {
+                    actionPlugins.add(criteria.getActionPlugin());
+                }
+                query.append(inClause("actionPlugin", actionPlugins));
+            }
+            if (criteria.hasEventIdCriteria()) {
+                Set<String> eventIds = new HashSet<>();
+                if (null != criteria.getEventIds()) {
+                    eventIds.addAll(criteria.getEventIds());
+                }
+                if (null != criteria.getEventId()) {
+                    eventIds.add(criteria.getEventId());
+                }
+                query.append(inClause("eventId", eventIds));
+            }
+            if (criteria.hasResultCriteria()) {
+                Set<String> results = new HashSet<>();
+                if (null != criteria.getResults()) {
+                    results.addAll(criteria.getResults());
+                }
+                if (null != criteria.getResult()) {
+                    results.add(criteria.getResult());
+                }
+                query.append(inClause("result", results));
+            }
+        }
+
+        List<IspnAction> ispnActions = queryFactory.create(query.toString()).list();
+        return prepareActionsPage(ispnActions.stream().map(a -> a.getAction()).collect(Collectors.toList()), pager);
+    }
+
+    // An exploded "in" clause because the actual one seems not to work
+    private String inClause(String field, Set<String> values) {
```

Comment:
I use the same pattern in alerts criteria.
I think you need an iterator here or you will have a " and ( or field = 'value' or field = 'value')" and this problably will throw a syntax error.
I used a 
```
Iterator<String> iter = values.iterator();
while (iter.hasNext()) {
      String value = iter.next();
      sb.append("field = '").append(value).append("'");
      if (iter.hasNext()) {
        sb.append(" or ");
      }
}
```